src/cargo/ops/lockfile.rs

```diff
@@ -19,20 +20,25 @@ pub fn load_pkg_lockfile(pkg: &Package) -> CargoResult<Option<Resolve>> {
 
 pub fn load_lockfile(path: &Path, sid: &SourceId) -> CargoResult<Option<Resolve>> {
     // If there is no lockfile, return none.
-    let mut f = match File::open(path) {
-        Ok(f) => f,
+    let s = match load_string(path) {
+        Ok(s) => s,
         Err(_) => return Ok(None)
     };
 
-    let mut s = String::new();
-    try!(f.read_to_string(&mut s));
-
     let table = toml::Value::Table(try!(cargo_toml::parse(&s, path)));
     let mut d = toml::Decoder::new(table);
     let v: resolver::EncodableResolve = try!(Decodable::decode(&mut d));
     Ok(Some(try!(v.to_resolve(sid))))
 }
 
+fn load_string(path: &Path) -> io::Result<String> {
```

Comment:
I believe the function `paths::read` also does the same thing (and should probably be used for the extra contextual error information)